modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/LZ4CompressionFactory.java

```diff
@@ -0,0 +1,171 @@
+        return Arrays.copyOf(compressed, compressedLength);
+    }
+
+    /**
+     * Decompress the compressed message content
+     *
+     * @param messagePartMap        Compressed message content as a andes message part map
+     * @param originalContentLength Total content length of the original message content
+     * @return Decompressed message content as a byte array
+     */
+    public byte[] getDecompressedMessage(Map<Integer, AndesMessagePart> messagePartMap, int originalContentLength) {
+        byte[] compressedMessageContent = getByteArrayFromMessagePartMap(messagePartMap, originalContentLength);
+
+        // Decompress message content
+        byte[] decompressedMessage = new byte[originalContentLength];
+        decompressor.decompress(compressedMessageContent, 0, decompressedMessage, 0, originalContentLength);
+
+        return decompressedMessage;
+    }
+
+    /**
+     * Decompress the compressed message content
+     *
+     * @param partList              Compressed message content, as a message part list
+     * @param originalContentLength Total content length of the original message content
+     * @return Decompressed message content as a byte array
+     */
+    public byte[] getDecompressedMessage(List<AndesMessagePart> partList, int originalContentLength) {
+        byte[] compressedMessageContent = getByteArrayFromPartList(partList, originalContentLength);
+
+        // Decompress message content
+        byte[] decompressedMessage = new byte[originalContentLength];
+        decompressor.decompress(compressedMessageContent, 0, decompressedMessage, 0, originalContentLength);
+
+        return decompressedMessage;
+    }
+
+    /**
+     * Make one byte array from data of andes message parts
+     *
+     * @param partList              Message content as a AndesMessagePart list
+     * @param originalContentLength Total content length of the above message content
+     * @return Combined message content as a byte array
+     */
+    private byte[] getByteArrayFromPartList(List<AndesMessagePart> partList, int originalContentLength) {
+        byte[] messageData = new byte[originalContentLength];
+        byte[] messagePartData;
+        int writingPosition = 0;
+
+        for (AndesMessagePart messagePart : partList) {
+            messagePartData = messagePart.getData();
+            System.arraycopy(messagePartData, 0, messageData, writingPosition, messagePartData.length);
+            writingPosition = writingPosition + messagePartData.length;
+        }
+
+        return messageData;
+    }
+
+    /**
+     * Make one byte array from data of andes message parts maps
+     *
+     * @param messagePartMap Message content as a map
+     * @param contentLength  Total content length of the above message content
+     * @return Combined message content as a byte array
+     */
+    private byte[] getByteArrayFromMessagePartMap(Map<Integer, AndesMessagePart> messagePartMap, int
+            contentLength) {
```

Comment:
Since the key of the input map is not used in this method, we should be able to get a list like 1D data structure here and do the processing.